run_route_scripts/run_with_server_matrix.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import argparse
+import multiprocessing
+from pathlib import Path
+
+import requests
+import json
+import time
+
+
+# generator for post bodies from the file
+def get_post_bodies(filename):
+    with open(filename, 'r') as f:
+        line_number = 0
+        for line in f:
+            line_number += 1
+            line = line[line.find('{'):]
+            line = line[0:line.rfind('}') + 1]
+            post_body = json.loads(line)
+            post_body['id'] = str(line_number)
+            yield post_body
+
+
+def initialize(args_, response_count_):
+    # for persistent connections
+    global session
+    session = requests.Session()
+    # so each process knows the options provided
+    global args
+    args = args_
+    # so each process can signal completing a request
+    global response_count
+    response_count = response_count_
+
+
+# post a request
+def make_request(post_body):
+    # make request
+    try:
+        start = time.time()
+        response = session.post(args.url, json=post_body, headers=args.headers)
+        stop = time.time()
+        elapsed = stop - start
+        response = response.json()
+        with response_count.get_lock():
+            response_count.value += 1
+    except Exception as e:
+        return {"error": e, "line": post_body["id"]}
+
+    return {"response": response, "line": post_body["id"], "perf": elapsed, "sources": post_body["sources"], "targets": post_body["targets"]}
+
+
+if __name__ == "__main__":
+    # parse some program arguments
+    parser = argparse.ArgumentParser(description="Runs payloads generated by gen_requests_matrix.py against a Valhalla server and writes the results to a single file for diffing purposes. Also prints out the total time.")
+    parser.add_argument('--test-file', type=Path, help='The file with the test requests', required=True)
+    parser.add_argument('--url', type=str, help='The url to which you want to POST the request bodies',
+                        default='http://localhost:8002/route')
```

Comment:
case in point haha

Reply:
Na the other script is doing one file per request, for matrix that’s not cool

Reply:
i was focused on this part

> default='http://localhost:8002/**route**'